rpc/src/rpc_service.rs

```diff
@@ -333,19 +312,12 @@ impl RequestMiddleware for RpcRequestMiddleware {
                     .body(hyper::Body::from(self.health_check()))
                     .unwrap()
                     .into(),
-                "/metrics" if self.enable_prometheus_metrics => {
-                    let banks_with_commitment =
-                        BanksWithCommitments::new(&self.bank_forks, &self.block_commitment_cache);
+                "/metrics" if self.prometheus_metrics.is_some() => {
+                    let prometheus_metrics = self.prometheus_metrics.as_ref().unwrap();
```

Comment:
If it was all our code, I would instead change the `match` to match on `(request.uri.path(), self.prometheus_metrics.as_ref())`, so we can match on `("/metrics", Some(prom))`, and we don’t have this `is_some` + `unwrap`.

But since this is a patch we are maintaining, an additional concern is to keep the diff as small as possible and try to not touch code paths that may change upstream. So I think the `is_some` + `unwrap` is the right approach here :+1: